orchagent/fdborch.h

```diff
@@ -5,29 +5,70 @@
 #include "observer.h"
 #include "portsorch.h"
 
+enum FdbOrigin
+{
+    FDB_ORIGIN_INVALID = 0,
+    FDB_ORIGIN_LEARN = 1,
+    FDB_ORIGIN_PROVISIONED = 2,
+    FDB_ORIGIN_VXLAN_ADVERTIZED = 4
+};
+
 struct FdbEntry
 {
     MacAddress mac;
     sai_object_id_t bv_id;
-    std::string port_name;
 
     bool operator<(const FdbEntry& other) const
     {
         return tie(mac, bv_id) < tie(other.mac, other.bv_id);
     }
+    bool operator==(const FdbEntry& other) const
+    {
+        return tie(mac, bv_id) == tie(other.mac, other.bv_id);
+    }
 };
 
 struct FdbUpdate
 {
     FdbEntry entry;
     Port port;
+    string type;
     bool add;
 };
 
+struct FdbData
+{
+    sai_object_id_t bridge_port_id;
+    string type;
+    FdbOrigin origin;
+    /**
+      {"dynamic", FDB_ORIGIN_LEARN} => dynamically learnt
+      {"dynamic", FDB_ORIGIN_PROVISIONED} => provisioned dynamic with swssconfig in APPDB
+      {"dynamic", FDB_ORIGIN_ADVERTIZED} => synced from remote device e.g. BGP MAC route
+      {"static", FDB_ORIGIN_LEARN} => Invalid
+      {"static", FDB_ORIGIN_PROVISIONED} => statically provisioned
+      {"static", FDB_ORIGIN_ADVERTIZED} => sticky synced from remote device
+    */
+
+    /* Remote FDB related info */
+    string remote_ip;
+    string    esi;
+    unsigned int vni;
+};
+
 struct SavedFdbEntry
 {
-    FdbEntry entry;
+    MacAddress mac;
```

Comment:
Why did we remove FdbEntry and added mac and vlanID explicitly. Can we keep FdbEntry and use its overloaded comparision operator to compare savedFdbEntry?

Reply:
SaveFdbEntry is used for storing an FDB which has valid vlan (vlan is created) but is waiting for the port to become the associated vlan's member.
now if vlan gets deleted, this entry is stuck in the saveFdbEntry;
later when user deletes the FDB, there is no way to compare it with bv_id (because that is no more there) hence changed to vlan-id so that delete request can be compared against FDBs in savedFdbEntry.